frontend/src/lib/components/Alerts/views/EditAlertModal.tsx

```diff
@@ -0,0 +1,256 @@
+import { LemonCheckbox, LemonInput, LemonSelect, SpinnerOverlay } from '@posthog/lemon-ui'
+import { useActions, useValues } from 'kea'
+import { Form, Group } from 'kea-forms'
+import { AlertStateIndicator } from 'lib/components/Alerts/views/ManageAlertsModal'
+import { MemberSelectMultiple } from 'lib/components/MemberSelectMultiple'
+import { TZLabel } from 'lib/components/TZLabel'
+import { UserActivityIndicator } from 'lib/components/UserActivityIndicator/UserActivityIndicator'
+import { IconChevronLeft } from 'lib/lemon-ui/icons'
+import { LemonButton } from 'lib/lemon-ui/LemonButton'
+import { LemonField } from 'lib/lemon-ui/LemonField'
+import { LemonModal } from 'lib/lemon-ui/LemonModal'
+import { alphabet } from 'lib/utils'
+import { trendsDataLogic } from 'scenes/trends/trendsDataLogic'
+
+import { AlertCalculationInterval, AlertType } from '~/queries/schema'
+import { QueryBasedInsightModel } from '~/types'
+
+import { alertFormLogic, AlertFormType } from '../alertFormLogic'
+import { alertLogic } from '../alertLogic'
+
+interface EditAlertProps {
+    isOpen: boolean | undefined
+    alertId?: string
+    insight?: Partial<QueryBasedInsightModel>
+    onEditSuccess: () => void
+    onClose?: () => void
+}
+
+export function AlertStateTable({ alert }: { alert: AlertType }): JSX.Element | null {
+    if (!alert.checks || alert.checks.length === 0) {
+        return null
+    }
+
+    return (
+        <div className="bg-bg-3000 p-4 mt-10 rounded-lg">
+            <h3>
+                Current status {alert.state}
+                <AlertStateIndicator alert={alert} />
+            </h3>
+            <table className="w-full table-auto border-spacing-2 border-collapse">
+                <thead>
+                    <tr className="text-left">
+                        <th>Status</th>
+                        <th className="text-right">Time</th>
+                        <th className="text-right pr-4">Value</th>
+                        <th>Targets notified</th>
+                    </tr>
+                </thead>
+                <tbody>
+                    {alert.checks.map((check) => (
+                        <tr key={check.id}>
+                            <td>{check.state}</td>
+                            <td className="text-right">
+                                <TZLabel time={check.created_at} />
+                            </td>
+                            <td className="text-right pr-4">{check.calculated_value}</td>
+                            <td>{check.targets_notified ? 'Yes' : 'No'}</td>
+                        </tr>
+                    ))}
+                </tbody>
+            </table>
+        </div>
+    )
+}
+
+export function EditAlertModal({ alertId, onClose, isOpen, onEditSuccess, insight }: EditAlertProps): JSX.Element {
+    if (!alertId && !insight) {
+        throw new Error('Must provide either an alertId or an insight to create alert for')
```

Comment:
Yes need to clean this up.
It was optional just for the scenario when they click on the alert from the `Alerts` tab on the saved insights page [here](https://github.com/PostHog/posthog/pull/25056/files#diff-ed56dd2c4b803d866af1cd1cf2dc41cdc48ec939c821210ae01b63c3f01514aeR83)

Is the best way to solve it to use the `alertId`
then find the alert from [here](https://github.com/PostHog/posthog/pull/25056/files#diff-ed56dd2c4b803d866af1cd1cf2dc41cdc48ec939c821210ae01b63c3f01514aeR24) 
and then pass the insight on the alert?

Reply:
Got it. You can add a loader when an insight is loading or serialize an insight in the alert if you have a foreign key (no waterfalls). You still need to load an insight to display the series dropdown, right? In that case, it would be better to show some UI feedback.